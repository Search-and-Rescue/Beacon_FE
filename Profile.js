import React, { Component } from "react";
import { View, StyleSheet, Text, Button } from "react-native";

export default class Profile extends Component {
  render() {
    return (
      <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
        <Text>Profile</Text>
      </View>
    );
  }
}
