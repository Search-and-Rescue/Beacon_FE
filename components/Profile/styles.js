const styles = {
  profileContainer: {
    flex: 1,
    marginVertical: 40,
    alignItems: "center"
  },
  modalToggleButton: {
    backgroundColor: '#efefef',
    marginVertical: 4,
    width: "80%"
  },
  modalToggleText: {
    fontSize: 21, 
    paddingVertical: 5,
    textAlign: "center"
  },
  profileFlatList: {
    width: "80%"
  },
  pickerView: {
    backgroundColor: "#efefef",
    margin: 5,
    padding: 20,
    position: "absolute",
    bottom: 0,
    left: 10,
    right: 10
  },
  modalHeading: {
    fontWeight: "bold",
    marginBottom: 5
  },
  modalButton: {
    paddingVertical: 10,
    alignItems: "center"
  },
  label: {
    fontSize: 12
  },
  input: {
    backgroundColor: "#fff",
    borderColor: "black",
    borderWidth: 1,
    marginBottom: 10,
    padding: 10
  },
  updateBtn: {
    borderColor: "black",
    borderWidth: 1,
    padding: 10,
    width: "80%"
  },
  updateBtnText: {
    textAlign: "center"
  }
};

export default styles;