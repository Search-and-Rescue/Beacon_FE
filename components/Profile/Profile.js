import React, { Component } from "react";
import { Button, 
  Modal,
  ScrollView, 
  View, 
  Text, 
  TextInput, 
  TouchableHighlight, 
  TouchableOpacity} from "react-native";
import styles from "./styles";

class Profile extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      address: "",
      city: "",
      state: "",
      zip: "",
      phone: "",
      email: "",
      password_digest: "",
      allergies: "",
      birthdate: "",
      weight: "",
      height: "",
      hair_color: "",
      skin_color: "",
      gender: "",
      experience_level: 0,
      experience_modal: false,
      cosar_card: false,
      cosar_modal: false
    };
  }

  handleSubmit = () => {
    const { navigation } = this.props;
    navigation.navigate("VehicleList");
    this.setState({
      name: "",
      address: "",
      city: "",
      state: "",
      zip: "",
      phone: "",
      email: "",
      password_digest: "",
      allergies: "",
      birthdate: "",
      weight: "",
      height: "",
      hair_color: "",
      skin_color: "",
      gender: "",
      experience_level: 0,
      experience_modal: false,
      cosar_card: false,
      cosar_modal: false
    });
  };

  setExperience = level => {
    this.setState({ experience_level: level });
    this.toggleExperienceModal();
  };

  toggleExperienceModal = () => {
    this.setState({ experience_modal: !this.state.experience_modal });
  };

  setCosar = bool => {
    this.setState({ cosar_card: bool });
    this.toggleCosarModal();
  };

  toggleCosarModal = () => {
    this.setState({ cosar_modal: !this.state.cosar_modal });
  };

  render() {
    const experienceLevels = [
      {
        title: "Beginner",
        value: 1
      },
      {
        title: "Intermediate",
        value: 2
      },
      {
        title: "Advanced",
        value: 3
      }
    ];

    const cosars = [
      {
        title: "Yes",
        value: true
      },
      {
        title: "No",
        value: false
      }
    ];

    return (
      <View style={styles.profileContainer}>
        <Button
          onPress={() => this.toggleExperienceModal()}
          title={"Choose Experience Level"}
        ></Button>
        <Button
          onPress={() => this.toggleCosarModal()}
          title={"Add COSAR Card Status"}
        ></Button>
        <Modal
          animationType={"slide"}
          visible={this.state.experience_modal}
          transparent={true}
          onRequestClose={() => console.log("close requested")}
        >
          <View style={styles.pickerView}>
            <Text style={styles.modalHeading}>Select Experience Level:</Text>
            {experienceLevels.map((level, index) => {
              return (
                <TouchableHighlight
                  key={index}
                  style={styles.modalButton}
                  onPress={() => this.setExperience(level.value)}
                >
                  <Text>{level.title}</Text>
                </TouchableHighlight>
              );
            })}
          </View>
        </Modal>
        <Modal
          animationType={"slide"}
          visible={this.state.cosar_modal}
          transparent={true}
          onRequestClose={() => console.log("close requested")}
        >
          <View style={styles.pickerView}>
            <Text style={styles.modalHeading}>
              Owner of current COSAR Card:
            </Text>
            {cosars.map((cosar, index) => {
              return (
                <TouchableHighlight
                  key={index}
                  style={styles.modalButton}
                  onPress={() => this.setCosar(cosar.value)}
                >
                  <Text>{cosar.title}</Text>
                </TouchableHighlight>
              );
            })}
          </View>
        </Modal>
        <ScrollView style={styles.profileFlatList}>
          <Text style={styles.label}>Name:</Text>
          <TextInput
            placeholder="Jill A. Hiker"
            style={styles.input}
            onChangeText={text => this.setState({ name: text })}
            value={this.state.name}
            name="name"
          />
          <Text style={styles.label}>Home address:</Text>
          <TextInput
            placeholder="123 Main St, Apt. 3"
            style={styles.input}
            onChangeText={text => this.setState({ address: text })}
            value={this.state.address}
            name="address"
          />
          <Text style={styles.label}>City:</Text>
          <TextInput
            placeholder="Denver"
            style={styles.input}
            onChangeText={text => this.setState({ city: text })}
            value={this.state.city}
            name="city"
          />
          <Text style={styles.label}>State:</Text>
          <TextInput
            placeholder="Colorado"
            style={styles.input}
            onChangeText={text => this.setState({ state: text })}
            value={this.state.state}
            name="state"
          />
          <Text style={styles.label}>Zip code:</Text>
          <TextInput
            keyboardType={"numeric"}
            placeholder="80202"
            style={styles.input}
            onChangeText={text => this.setState({ zip: text })}
            value={`${this.state.zip}`}
            name="zip"
          />
          <Text style={styles.label}>Phone number:</Text>
          <TextInput
            keyboardType={"phone-pad"}
            placeholder="(970) 123-4567"
            style={styles.input}
            onChangeText={text => this.setState({ phone: text })}
            value={this.state.phone}
            name="phone"
          />
          <Text style={styles.label}>Email address:</Text>
          <TextInput
            keyboardType={"email-address"}
            placeholder="me@email.com"
            style={styles.input}
            onChangeText={text => this.setState({ email: text })}
            value={this.state.email}
            name="email"
          />
          <Text style={styles.label}>Password:</Text>
          <TextInput
            placeholder="********"
            style={styles.input}
            onChangeText={text => this.setState({ password_digest: text })}
            value={this.state.password_digest}
            name="password_digest"
          />
          <Text style={styles.label}>My allergies:</Text>
          <TextInput
            placeholder="Penicillin, peanuts, latex"
            style={styles.input}
            onChangeText={text => this.setState({ allergies: text })}
            value={this.state.allergies}
            name="allergies"
          />
          <Text style={styles.label}>My outdoor experience:</Text>
          <TextInput
            keyboardType={"numeric"}
            placeholder="Beginner, Intermediate, Advanced"
            style={styles.input}
            onChangeText={text =>
              this.setState({ experience_level: Number(text) })
            }
            value={`${this.state.experience_level}`}
            name="experience_level"
          />
          <Text style={styles.label}>:</Text>
          <TextInput
            placeholder="Birthdate"
            style={styles.input}
            onChangeText={text => this.setState({ birthdate: text })}
            value={this.state.birthdate}
            name="birthdate"
          />
          <Text style={styles.label}>:</Text>
          <TextInput
            keyboardType={"numeric"}
            placeholder="Weight"
            style={styles.input}
            onChangeText={text => this.setState({ weight: Number(text) })}
            value={`${this.state.weight}`}
            name="weight"
          />
          <Text style={styles.label}>:</Text>
          <TextInput
            placeholder="Height"
            style={styles.input}
            onChangeText={text => this.setState({ height: text })}
            value={this.state.height}
            name="height"
          />
          <Text style={styles.label}>:</Text>
          <TextInput
            placeholder="Hair color"
            style={styles.input}
            onChangeText={text => this.setState({ hair_color: text })}
            value={this.state.hair_color}
            name="hair_color"
          />
          <Text style={styles.label}>:</Text>
          <TextInput
            placeholder="Skin color"
            style={styles.input}
            onChangeText={text => this.setState({ skin_color: text })}
            value={this.state.skin_color}
            name="skin_color"
          />
          <Text style={styles.label}>:</Text>
          <TextInput
            placeholder="Gender"
            style={styles.input}
            onChangeText={text => this.setState({ gender: text })}
            value={this.state.gender}
            name="gender"
          />
          <Text style={styles.label}>:</Text>
          <TextInput
            placeholder="COSAR card"
            style={styles.input}
            onChangeText={text => this.setState({ COSAR_card: Boolean(text) })}
            value={`${this.state.COSAR_card}`}
            name="COSAR_card"
          />
          <Text style={styles.label}>:</Text>
        </ScrollView>
        <TouchableOpacity style={styles.updateBtn} onPress={this.handleSubmit}>
          <Text style={styles.updateBtnText}>Update</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default Profile;