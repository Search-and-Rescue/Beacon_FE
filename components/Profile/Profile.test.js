import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Profile from './Profile';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Profile', () => {
  it('should match the snapshot with all of the data passed through', () => {
    expect(shallow(<Profile />)).toMatchSnapshot();
  });
});