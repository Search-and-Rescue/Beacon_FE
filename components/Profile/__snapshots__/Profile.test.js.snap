// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match the snapshot with all of the data passed through 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginVertical": 40,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#efefef",
          "marginVertical": 4,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 21,
            "paddingVertical": 5,
            "textAlign": "center",
          }
        }
      >
        Select Experience Level
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#efefef",
          "marginVertical": 4,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 21,
            "paddingVertical": 5,
            "textAlign": "center",
          }
        }
      >
        Select COSAR Card Status
      </Text>
    </TouchableOpacity>
    <Component
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "backgroundColor": "#efefef",
            "bottom": 0,
            "left": 10,
            "margin": 5,
            "padding": 20,
            "position": "absolute",
            "right": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Select Experience Level:
        </Text>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          key="0"
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "paddingVertical": 10,
            }
          }
          underlayColor="black"
        >
          <Text>
            Beginner
          </Text>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          key="1"
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "paddingVertical": 10,
            }
          }
          underlayColor="black"
        >
          <Text>
            Intermediate
          </Text>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          key="2"
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "paddingVertical": 10,
            }
          }
          underlayColor="black"
        >
          <Text>
            Advanced
          </Text>
        </TouchableHighlight>
      </View>
    </Component>
    <Component
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "backgroundColor": "#efefef",
            "bottom": 0,
            "left": 10,
            "margin": 5,
            "padding": 20,
            "position": "absolute",
            "right": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
              "marginBottom": 5,
            }
          }
        >
          Owner of current COSAR Card:
        </Text>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          key="0"
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "paddingVertical": 10,
            }
          }
          underlayColor="black"
        >
          <Text>
            Yes
          </Text>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          key="1"
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "paddingVertical": 10,
            }
          }
          underlayColor="black"
        >
          <Text>
            No
          </Text>
        </TouchableHighlight>
      </View>
    </Component>
    <ScrollViewMock
      style={
        Object {
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Name:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="name"
        onChangeText={[Function]}
        placeholder="Jill A. Hiker"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Home address:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="address"
        onChangeText={[Function]}
        placeholder="123 Main St, Apt. 3"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        City:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="city"
        onChangeText={[Function]}
        placeholder="Denver"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        State:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="state"
        onChangeText={[Function]}
        placeholder="Colorado"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Zip code:
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        name="zip"
        onChangeText={[Function]}
        placeholder="80202"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Phone number:
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="phone-pad"
        name="phone"
        onChangeText={[Function]}
        placeholder="(970) 123-4567"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Email address:
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="email-address"
        name="email"
        onChangeText={[Function]}
        placeholder="jill@email.com"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Birthday:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="birthdate"
        onChangeText={[Function]}
        placeholder="October 1, 1985"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Password:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="password_digest"
        onChangeText={[Function]}
        placeholder="********"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Allergies:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="allergies"
        onChangeText={[Function]}
        placeholder="Penicillin, peanuts, latex"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Height:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="height"
        onChangeText={[Function]}
        placeholder="5 08"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Weight:
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        name="weight"
        onChangeText={[Function]}
        placeholder="160"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Hair Color:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="hair_color"
        onChangeText={[Function]}
        placeholder="Blonde"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Skin Color:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="skin_color"
        onChangeText={[Function]}
        placeholder="Medium"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "textAlign": "left",
          }
        }
      >
        Gender:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="gender"
        onChangeText={[Function]}
        placeholder="F"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </ScrollViewMock>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "padding": 10,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Update
      </Text>
    </TouchableOpacity>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
</KeyboardAvoidingView>
`;
